src/NServiceBus.Transport.SQS.AcceptanceTests/When_trying_to_subscribe_when_topic_does_not_exist.cs

```diff
@@ -0,0 +1,123 @@
+﻿namespace NServiceBus.AcceptanceTests
+{
+    using System;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using NServiceBus.AcceptanceTesting.Customization;
+    using NServiceBus.AcceptanceTesting.Support;
+    using NServiceBus.Features;
+    using NUnit.Framework;
+
+    public class When_trying_to_subscribe_when_topic_does_not_exist : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_log_exception_with_topic_name()
+        {
+            var messageId = Guid.NewGuid();
+
+            await Scenario.Define<Context>(c =>
+            {
+                c.MessageId = messageId;
+                c.EnableInstallers = true;
+                c.DisableAutoSubscribe = true;
+            })
+            .WithEndpoint<Endpoint>()
+            .Done(context => true)
+            .Run();
+
+            var context = await Scenario.Define<Context>(c =>
+            {
+                c.MessageId = messageId;
+                c.EnableInstallers = false;
+                c.DisableAutoSubscribe = false;
+            })
+            .WithEndpoint<Endpoint>()
+            .Done(context => true)
+            .Run();
+
+            Assert.IsTrue(context.Logs.Any(a => a.Message.Contains("not found") && a.Message.Contains("When_trying_to_subscribe_when_topic_does_not_exist-MyEvent")));
+        }
+
+        public class Context : ScenarioContext
+        {
+            public Guid MessageId { get; set; }
+            public bool EnableInstallers { get; set; }
+            public bool DisableAutoSubscribe { get; set; }
+        }
+
+        public class Endpoint : EndpointConfigurationBuilder
+        {
+            public Endpoint() =>
+                EndpointSetup<DefaultServerWithNoInstallers>((e, c) =>
+                {
+                    var ourContext = c.ScenarioContext as Context;
+                    if (ourContext.EnableInstallers)
+                    {
+                        e.EnableInstallers();
+                    }
+                    if (ourContext.DisableAutoSubscribe)
+                    {
+                        e.DisableFeature<AutoSubscribe>();
+                    }
```

Comment:
Isn't this plus the two scenario define calls an indication we should split into two Shoulds or potentially even whens?

Reply:
What we need here is for the topic to not be created - we want to test the error message when someone is subscribing to a topic that doesn't exist.
 In order to test that, we need the installers off. However, when that happens, an exception will be thrown earlier on in the pipeline during the `InputQueuePump.Initialize` call, before we can test our subscription.
So what we are doing is creating the first endpoint with installers but without auto subscribe so that `SubscribeAll `is not called and hence the topic is not created.
We then create a second endpoint that has the installers disabled but subscibers on. So, it will not throw the exception during the `InputQueuePump.Initialize` call since the queue was created by the previous endpoint, but it will at the `SubscribeAll` point since the topic does not exist and installers are off.

From what I can see we can't call a `.WithEndpoint` based on a `When` - ie don't start the second endpoint until the first one is started. But I could be wrong!